File: CS/PopupSizeExample.Module.Web/Controllers/PopupWindowUpdateSizeViewController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Web.UI;
using DevExpress.ExpressApp.Web;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Web.Controls;
using System.Drawing;
using PopupSizeExample.Module.BusinessObjects;

namespace PopupSizeExample.Module.Web.Controllers {
    public class PopupWindowUpdateSizeViewController : ViewController {
        protected override void OnViewControlsCreated() {
            base.OnViewControlsCreated();
            if (WebWindow.CurrentRequestPage != null) {
                WebWindow.CurrentRequestPage.Load += new EventHandler(CurrentRequestPage_Load);
                WebWindow.CurrentRequestPage.Unload += new EventHandler(CurrentRequestPage_Unload);
            }
        }
        private void CurrentRequestPage_Load(object sender, EventArgs e) {
            IPopupWindowControlHolder holder = sender as IPopupWindowControlHolder;
            if (holder != null && holder.PopupWindowControl != null) {
                holder.PopupWindowControl.CustomizePopupWindowSize += 
                    PopupWindowControl_CustomizePopupControlSize;
            }
        }
        private void CurrentRequestPage_Unload(object sender, EventArgs e) {
            Page page = (Page)sender;
            page.Load -= CurrentRequestPage_Load;
            page.Unload -= CurrentRequestPage_Unload;
            IPopupWindowControlHolder holder = page as IPopupWindowControlHolder;
            if (holder != null && holder.PopupWindowControl != null) {
                holder.PopupWindowControl.CustomizePopupWindowSize -= 
                    PopupWindowControl_CustomizePopupControlSize;
            }
        }
        private void PopupWindowControl_CustomizePopupControlSize(object sender, CustomizePopupWindowSizeEventArgs e) {
            PopupWindow window;
            try { window = e.FindPopupWindow((WebApplication)Application); }
            catch (ArgumentException) { return; }
            if(window != null && window.View != null) {
                PopupInfoDemoObject currentObjectInPopup = window.View.CurrentObject as PopupInfoDemoObject;
                PopupDemoObject currentParentObject = View.CurrentObject as PopupDemoObject;
                if ((currentParentObject != null) && (currentObjectInPopup != null)) {
                    e.Size = new Size(currentParentObject.Width, currentParentObject.Height);
                    currentObjectInPopup.Info = string.Format("This popup size is {0}x{1}",
                        currentParentObject.Width, currentParentObject.Height);
                    e.Handled = true;
                }
            }
        }
    }
    public interface IPopupWindowControlHolder {
        XafPopupWindowControl PopupWindowControl { get; }
    }
}

File: VB/PopupSizeExample.Module.Web/Controllers/PopupWindowUpdateSizeViewController.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Web.UI
Imports DevExpress.ExpressApp.Web
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Web.Controls
Imports System.Drawing
Imports PopupSizeExample.Module.BusinessObjects

Namespace PopupSizeExample.Module.Web.Controllers
	Public Class PopupWindowUpdateSizeViewController
		Inherits ViewController
		Protected Overrides Sub OnViewControlsCreated()
			MyBase.OnViewControlsCreated()
			If WebWindow.CurrentRequestPage IsNot Nothing Then
				AddHandler WebWindow.CurrentRequestPage.Load, AddressOf CurrentRequestPage_Load
				AddHandler WebWindow.CurrentRequestPage.Unload, AddressOf CurrentRequestPage_Unload
			End If
		End Sub
		Private Sub CurrentRequestPage_Load(ByVal sender As Object, ByVal e As EventArgs)
			Dim holder As IPopupWindowControlHolder = TryCast(sender, IPopupWindowControlHolder)
			If holder IsNot Nothing AndAlso holder.PopupWindowControl IsNot Nothing Then
				AddHandler holder.PopupWindowControl.CustomizePopupWindowSize, AddressOf PopupWindowControl_CustomizePopupControlSize
			End If
		End Sub
		Private Sub CurrentRequestPage_Unload(ByVal sender As Object, ByVal e As EventArgs)
			Dim page As Page = CType(sender, Page)
			RemoveHandler page.Load, AddressOf CurrentRequestPage_Load
			RemoveHandler page.Unload, AddressOf CurrentRequestPage_Unload
			Dim holder As IPopupWindowControlHolder = TryCast(page, IPopupWindowControlHolder)
			If holder IsNot Nothing AndAlso holder.PopupWindowControl IsNot Nothing Then
				RemoveHandler holder.PopupWindowControl.CustomizePopupWindowSize, AddressOf PopupWindowControl_CustomizePopupControlSize
			End If
		End Sub
		Private Sub PopupWindowControl_CustomizePopupControlSize(ByVal sender As Object, ByVal e As CustomizePopupWindowSizeEventArgs)
			Dim window As PopupWindow
			Try
 			       window = e.FindPopupWindow(CType(Application, WebApplication))
    			Catch e1 As ArgumentException
                       			 Return
			End Try
			If window IsNot Nothing AndAlso window.View IsNot Nothing Then
				Dim currentObjectInPopup As PopupInfoDemoObject = TryCast(window.View.CurrentObject, PopupInfoDemoObject)
				Dim currentParentObject As PopupDemoObject = TryCast(View.CurrentObject, PopupDemoObject)
				If (currentParentObject IsNot Nothing) AndAlso (currentObjectInPopup IsNot Nothing) Then
					e.Size = New Size(currentParentObject.Width, currentParentObject.Height)
					currentObjectInPopup.Info = String.Format("This popup size is {0}x{1}", currentParentObject.Width, currentParentObject.Height)
					e.Handled = True
				End If
			End If
		End Sub
	End Class
	Public Interface IPopupWindowControlHolder
		ReadOnly Property PopupWindowControl() As XafPopupWindowControl
	End Interface
End Namespace
